### PR TITLE
feat: update to Go 1.21

### PR DESCRIPTION
**Pull request by ksimon1**
**What this PR does / why we need it**:
Update all necessary files to use golang 1.21

**Release note**:
```
bump golang version to 1.21
```

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: 1.19
+          go-version: 1.21
```
Comment (0xFelix) [inline .github/workflows/release.yaml:18]: Make this read the go version from `go.mod` file instead?

Comment (ksimon1) [reply .github/workflows/release.yaml:18]: Correct, I will fix it

**Review by 0xFelix — APPROVED** [review]
/lgtm

**Pull request merged**